mods/taskbar-disappearing-icon-fix-win11.wh.cpp

```diff
@@ -0,0 +1,280 @@
+    wcscpy_s(path, MAX_PATH, szWindowsDirectory);
+    wcscat_s(
+        path, MAX_PATH,
+        LR"(\SystemApps\MicrosoftWindows.Client.Core_cw5n1h2txyewy\Taskbar.View.dll)");
+    if (GetFileAttributes(path) != INVALID_FILE_ATTRIBUTES) {
+        return true;
+    }
+
+    // Windows 11 version 21H2.
+    wcscpy_s(path, MAX_PATH, szWindowsDirectory);
+    wcscat_s(
+        path, MAX_PATH,
+        LR"(\SystemApps\MicrosoftWindows.Client.CBS_cw5n1h2txyewy\ExplorerExtensions.dll)");
+    if (GetFileAttributes(path) != INVALID_FILE_ATTRIBUTES) {
+        return true;
+    }
+
+    return false;
+}
+
+// ===========================================================
+
+void RefreshTaskbar() {
+    HWND hTaskbarWnd = GetTaskbarWnd();
+    if (!hTaskbarWnd) {
+        return;
+    }
+
+    SendMessage(hTaskbarWnd, WM_SETTINGCHANGE, 0, 0);
+    Sleep(400);
+    SendMessage(hTaskbarWnd, WM_SETTINGCHANGE, 0, 0);
+}
+
+FrameworkElement FindChildByName(FrameworkElement element, PCWSTR name) {
+    int childrenCount = Media::VisualTreeHelper::GetChildrenCount(element);
+
+    for (int i = 0; i < childrenCount; i++) {
+        auto child = Media::VisualTreeHelper::GetChild(element, i)
+                         .try_as<FrameworkElement>();
+        if (!child) {
+            Wh_Log(L"Failed to get child %d of %d", i + 1, childrenCount);
+            continue;
+        }
+
+        if (child.Name() == name) {
+            return child;
+        }
+    }
+
+    return nullptr;
+}
+
+// ===========================================================
+
+void HideAndShowIcon(FrameworkElement taskListButtonElement) {
+    if(g_unloading) {
+        return;
+    }
+    
+    auto iconPanelElement =
+        FindChildByName(taskListButtonElement, L"IconPanel");
+    if (!iconPanelElement) {
+        return;
+    }
+
+    auto iconElement =
+        FindChildByName(iconPanelElement, L"Icon")
+            .as<Controls::Image>();
+
+    if(iconElement) {
+        iconElement.Visibility(Visibility::Collapsed);
+        iconElement.Dispatcher().TryRunAsync(winrt::Windows::UI::Core::CoreDispatcherPriority::Low, [iconElement](){
+            iconElement.Visibility(Visibility::Visible);
+        });
+    }
+}
+
+using TaskListButton_Hook2_t = void(WINAPI*)(void* pThis, struct winrt::hstring w);
+TaskListButton_Hook2_t TaskListButton_Hook2_Original;
+void WINAPI TaskListButton_Hook2_Hook(void* pThis, struct winrt::hstring w) {
+    TaskListButton_Hook2_Original(pThis, w);
+
+    Wh_Log(L">Hook2");
+
+    void* taskListButtonIUnknownPtr = (void**)pThis + 3;
+    winrt::Windows::Foundation::IUnknown taskListButtonIUnknown;
+    winrt::copy_from_abi(taskListButtonIUnknown, taskListButtonIUnknownPtr);
+
+    auto taskListButtonElement = taskListButtonIUnknown.as<FrameworkElement>();
+
+    HideAndShowIcon(taskListButtonElement);
+}
+
+// ===========================================================
+
+bool HookTaskbarViewDllSymbols(HMODULE module) {
+    // taskbar.view.dll
```

Comment:
```suggestion
    // taskbar.view.dll, ExplorerExtensions.dll
```

Reply:
I am just wondering why is ExplorerExtensions.dll also considered a target. The symbols I use are found in taskbar.view.dll.

Reply:
Windows 11 version 22H2 had ExplorerExtensions.dll instead of taskbar.view.dll.